python/ray/train/__init__.py

```diff
@@ -2,14 +2,32 @@
 from ray.train.backend import BackendConfig
 from ray.train.data_config import DataConfig
 from ray.train.constants import TRAIN_DATASET_KEY
+from ray.train._internal.session import (
+    get_dataset_shard,
+    get_trial_resources,
+    get_world_rank,
+    get_world_size,
+    report,
+)
 from ray.train.trainer import TrainingIterator
 
+from ray.air import Checkpoint
+from ray.air.config import CheckpointConfig, RunConfig, ScalingConfig
 
 usage_lib.record_library_usage("train")
 
 __all__ = [
     "BackendConfig",
+    "Checkpoint",
+    "CheckpointConfig",
     "DataConfig",
+    "RunConfig",
+    "ScalingConfig",
     "TrainingIterator",
     "TRAIN_DATASET_KEY",
+    "get_dataset_shard",
+    "get_trial_resources",
+    "get_world_rank",
+    "get_world_size",
```

Comment:
Should these accessors be part of session only?

Reply:
I want to get rid of the `session` namespace going forward. It serves no purpose.

Reply:
I feel like we should do that atomically then, it seems odd to move just a few specific methods initially. It is also a big change and I'm not clear if the pros outweigh the cons.

Reply:
So I don't think `get_local_rank` and `get_local_world_size` are currently being used anywhere (they are internal APIs and used to automatically set an environment variable for TorchTrainer). So I'd like to get rid of them. Should I be wrong I'm happy to add them later. So is `get_node_rank` (which is also very confusing btw). Btw, the docs strings for all of these except `get_local_rank` is wrong too.

For `get_trial_dir`, `get_trial_id`, `get_trial_name`, `get_experiment_name` I would first like to understand how your checkpointing changes are changing things, but I'm happy to add them back if you think they will keep being the same.

`get_checkpoint` will probably become irrelevant.

That's already all the methods in `session`.

Reply:
So these are all important utility methods. The more general concern is that there will always be more and more of these small things, just see ray.get_runtime_context() for example (I believe session should be renamed something like context, IMO).

We should plan for a large number of utility methods existing here in the future, and I don't think it's great to have them directly scoped under `train`.

Reply:
Yeah, my ideal API for this would be
```
def train_loop_per_worker(context):
    # Use context.report(...), context.get_dataset_shard(...) etc.
```
Let me look into whether this can be made to happen in a compatible way.

Reply:
+1 to renaming `Session` to `*Context`.

I don't really see a benefit of passing it into the `train_loop_per_worker` or `trainable`, so I would prefer not to unless there is a strong enough reason for this.

Reply:
Alright, the plan changed to renaming `from ray.air import session` ->  `from ray.train import context` :)